### PR TITLE
Add primary flag for touch events to handle multi-touch scenarios

### PR DESCRIPTION
**Pull request by nairmanu**

### CODE REVIEW

File: packages/clarity-js/src/interaction/pointer.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ import encode from "./encode";
 
 export let state: PointerState[] = [];
 let timeout: number = null;
+let activeTouchPointId = 0;
+let activeTouchPointIds = new Set<number>();
```
Comment (malaref) [inline packages/clarity-js/src/interaction/pointer.ts:15]: nit: this can be `const`

**Review by malaref — APPROVED** [review]

File: packages/clarity-js/src/interaction/pointer.ts (hunk 1 above)
Comment (nairmanu) [reply packages/clarity-js/src/interaction/pointer.ts:15]: the value changes right? how can it be const?

Comment (ender336) [reply packages/clarity-js/src/interaction/pointer.ts:15]: const for a set means that the pointer doesn't change essentially, it doens't mean the contents of the set are the same (like an array you can have const x = []; x.push(1); x.push(2); //you just can't then do x = [2,3,4]

File: packages/clarity-js/src/interaction/pointer.ts [hunk 2]
```diff
@@ -58,13 +60,26 @@ function touch(event: Event, root: Node, evt: TouchEvent): void {
...
+                        activeTouchPointId = id;
+                    }
+                    activeTouchPointIds.add(id);
+                    break;
+                case Event.TouchEnd:
+                case Event.TouchCancel:
+                    activeTouchPointIds.delete(id);
+                    break;
+            }
+            const isPrimary: boolean = activeTouchPointId === id;
```
Comment (ender336) [inline packages/clarity-js/src/interaction/pointer.ts:79]: nit, you don't need to define the type here, it's implicitly defined by activeTouchPointId === id

File: packages/clarity-js/src/interaction/pointer.ts [hunk 3]
```diff
@@ -58,13 +60,26 @@ function touch(event: Event, root: Node, evt: TouchEvent): void {
...
+
             // Check for null values before processing this event
-            if (x !== null && y !== null) { handler({ time: t, event, data: { target: target(evt), x, y, id } }); }
+            if (x !== null && y !== null) { handler({ time: t, event, data: { target: target(evt), x, y, id, isPrimary } }); }
```
Comment (ender336) [inline packages/clarity-js/src/interaction/pointer.ts:82]: don't you need to add something into encode so this value actually gets sent?

File: packages/clarity-js/src/interaction/pointer.ts (hunk 2 above)
Comment (nairmanu) [reply packages/clarity-js/src/interaction/pointer.ts]: updated

File: packages/clarity-js/src/interaction/pointer.ts (hunk 1 above)
Comment (nairmanu) [reply packages/clarity-js/src/interaction/pointer.ts]: changed to const

File: packages/clarity-js/src/interaction/pointer.ts (hunk 3 above)
Comment (nairmanu) [reply packages/clarity-js/src/interaction/pointer.ts:82]: yes! added them. thanks!

**Review by ender336 — APPROVED** [review]

**Pull request merged**